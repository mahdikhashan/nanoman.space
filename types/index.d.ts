import { type Photo } from "react-photo-album"

type Project = {
  id: number;
  name: string;
  icon: string;
  slug: string;
  github: string;
  website: string;
  description: string;
  skills: string[];
  featured: boolean;
  type: string;
};

type HeaderItem = {
  name: string;
  slug: string;
  isDisabled?: boolean;
  redirect?: boolean;
  items?: {
    name: string;
    slug: string;
    description?: string;
    isDisabled?: boolean;
  }[];
};

type PhotoImage = Photo
