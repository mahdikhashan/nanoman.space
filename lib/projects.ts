type Project = {
  id: number;
  name: string;
  icon: string;
  slug: string;
  github: string;
  website: string;
  description: string;
  skills: string[];
  featured: boolean;
  type: string;
};

const projects: Project[] = [
  {
    id: 1,
    name: '99 Emails',
    icon: '/logo.svg',
    slug: '99-minutes-emails',
    github: 'https://github.com/mahdikhashan/99-minutes-emails',
    website: 'https://99-minutes-emails.vercel.app/',
    description: 'A user friendly, safe and available temporary email service.',
    skills: [
      'react',
      'typescript',
      'redux-saga',
      'chakra-ui',
      'vite',
      'vitest',
      'cypress',
      'storybook'
    ],
    featured: true,
    type: 'Side Project'
  },
  {
    id: 2,
    name: 'Aparat Python API',
    icon: '/package.png',
    slug: 'aparat-python',
    github: 'https://github.com/mahdikhashan/aparat-python',
    website: 'https://pypi.org/project/aparat-python/',
    description: 'Aparat VOD Platform API Wrapper',
    skills: ['python', 'REST-API', 'unit-test'],
    featured: true,
    type: 'Python Package'
  },
  {
    id: 3,
    name: 'Vue 3 Click Outside',
    icon: '/package.png',
    slug: 'vue3-click-outside',
    github: 'https://github.com/mahdikhashan/vue3-click-outside',
    website: 'https://www.npmjs.com/package/@mahdikhashan/vue3-click-outside',
    description:
      'Directive for Vue 3 to run a method on clicking outside of the binded element',
    skills: ['javascript', 'vue3', 'vue-test-utils', 'jest'],
    featured: true,
    type: 'NPM Package'
  },
  {
    id: 4,
    name: 'Vercel CORS Proxy',
    icon: '/package.png',
    slug: 'vercel-express-proxy',
    github: 'https://github.com/mahdikhashan/vercel-express-proxy',
    website: 'https://github.com/mahdikhashan/vercel-express-proxy',
    description: 'ðŸªž Proxy server based on Express for Vercel platform',
    type: 'Template / CLI',
    featured: false,
    skills: ['javascript', 'node.js', 'express.js', 'vercel']
  },
  {
    id: 5,
    name: 'Vercel Express',
    icon: '/package.png',
    slug: 'vercel-express-template',
    github: 'https://github.com/mahdikhashan/vercel-express-template',
    website: 'https://github.com/mahdikhashan/vercel-express-template',
    description: 'ðŸ”¥ Express Server Template for Vercel',
    type: 'Template / CLI',
    featured: false,
    skills: ['javascript', 'node.js', 'express.js', 'vercel']
  },
  {
    id: 6,
    name: 'Veemoji',
    icon: '/package.png',
    slug: 'veemoji',
    github: 'https://github.com/mahdikhashan/veemoji',
    website: 'https://veemoji.surge.sh/',
    description: 'A dead simple emoji component library',
    type: 'NPM Package',
    featured: true,
    skills: ['javascript', 'vue', 'npm']
  }
];

export { projects, type Project };
