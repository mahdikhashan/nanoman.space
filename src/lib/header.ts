type Item = {
  name: string;
  slug: string;
  isDisabled?: boolean;
  redirect?: boolean;
  items?: {
    name: string;
    slug: string;
    description?: string;
    isDisabled?: boolean;
  }[];
};

export const header: Item[] = [
  {
    name: 'Home',
    slug: '/'
  },
  {
    name: 'About',
    slug: '/about'
  },
  {
    name: 'Projects',
    slug: '/projects',
    items: [
      {
        name: 'Hooks',
        slug: 'hooks',
        description:
          'Preview the hooks available for Client and Server Components',
      },
      {
        name: 'Client Context',
        slug: 'context',
        description:
          'Pass context between Client Components that cross Server/Client Component boundary',
      },
    ],
  },
  {
    name: 'LinkedIn',
    slug: 'https://www.linkedin.com/in/mahdikhashan/',
    redirect: true,
    isDisabled: true
  },
  {
    name: 'Github',
    slug: 'https://github.com/mahdikhashan',
    redirect: true,
    isDisabled: true
  },
  {
    name: 'Twitter',
    slug: 'twitter',
    redirect: true,
    isDisabled: true
  },
];
