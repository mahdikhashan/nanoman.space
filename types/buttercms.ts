export type Post = {
  status: string,
  created: string,
  updated: string,
  published: string,
  title: string,
  slug: string,
  body: string,
  summary: string,
  seo_title: string,
  meta_description: string,
  featured_image_alt: string,
  url: string,
  featured_image: unknown,
  tags: Tag[],
  categories: Category[],
  author: Author
};

export type Category = {
  name: string,
  slug: string
};

export type Author = {
  bio: string,
  email: string,
  slug: string,
  title: string,
  last_name: string,
  first_name: string,
  facebook_url: string,
  linkedin_url: string,
  instagram_url: string,
  pinterest_url: string,
  profile_image: string,
  twitter_handle: string
};

export type Tag = {};

export type Views = {
  total: number;
};