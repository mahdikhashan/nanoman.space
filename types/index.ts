import { type Photo } from "react-photo-album"

export type Category = { name: string }
export type Post = {
  title: string,
  slug: string,
  categories: Category[]
}

export type Project = {
  id: number;
  name: string;
  icon: string;
  slug: string;
  github: string;
  website: string;
  description: string;
  skills: string[];
  featured: boolean;
  type: string;
};

export type Experience = {
  id: number;
  company: string;
  icon?: string;
  position: string;
  startDate: string;
  finishDate: string;
  url?: string;
}

export type Skill = {
  id: number;
  category: string;
  tools: string[];
}

export type HeaderItem = {
  name: string;
  slug: string;
  isDisabled?: boolean;
  redirect?: boolean;
  items?: {
    name: string;
    slug: string;
    description?: string;
    isDisabled?: boolean;
  }[];
};

export type PhotoImage = Photo
