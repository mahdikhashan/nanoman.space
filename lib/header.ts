import { HeaderItem } from "@/types";

export const header: HeaderItem[] = [
  {
    name: 'Home',
    slug: '',
    isDisabled: true,
  },
  {
    name: 'About',
    slug: 'about',
    isDisabled: true
  },
  {
    name: 'Blog',
    slug: 'blog',
    isDisabled: true
  },
  {
    name: 'Projects',
    slug: 'projects',
    isDisabled: true,
    items: [
      {
        name: 'Hooks',
        slug: 'hooks',
        description:
          'Preview the hooks available for Client and Server Components',
      },
      {
        name: 'Client Context',
        slug: 'context',
        description:
          'Pass context between Client Components that cross Server/Client Component boundary',
      },
    ],
  },
  {
    name: 'Travel',
    slug: 'travels',
    isDisabled: true
  },
  {
    name: 'LinkedIn',
    slug: 'https://www.linkedin.com/in/mahdikhashan/',
    redirect: true,
    isDisabled: true
  },
  {
    name: 'Github',
    slug: 'https://github.com/mahdikhashan',
    redirect: true,
    isDisabled: true
  },
  {
    name: 'Twitter',
    slug: 'https://twitter.com/__nanoman__',
    redirect: true,
    isDisabled: true
  },
];
