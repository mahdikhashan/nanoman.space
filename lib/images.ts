import { PhotoImage } from "@/types";

export const images: PhotoImage[] = [
  { src: '/images/me_in_laft.jpeg', width: 400, height: 200, alt: "me" },
  { src: '/images/me_in_laft_w.jpeg', width: 400, height: 800, alt: "me" },
  { src: '/images/me_in_laft_t.jpeg', width: 200, height: 100, alt: "me" },
];
