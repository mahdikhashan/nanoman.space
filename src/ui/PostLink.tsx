'use client';

import { Post } from '@/lib/types';
import Link from 'next/link';

export type Props = Pick<Post, 'slug' | 'title' | 'categories'>;

export default function PostLink({
  slug,
  title,
  categories
}: Props) {
  return (
    <Link
      href={`/blog/${slug}`}
      className="flex flex-row py-4 justify-between hover:text-orange-500 transition ease-in-out delay-50 hover:scale-[1.025]"
    >
      <div>{title}</div>
      <div className="text-zinc-500">
        {categories.map((category: any) => category.name)}
      </div>
    </Link>
  );
}
