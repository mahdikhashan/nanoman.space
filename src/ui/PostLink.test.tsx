import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom'

import PostLink from './PostLink';
import type { Props as PostLinkProps } from './PostLink';
import { Category } from '@/lib/types';

const categories: Category[] = 
  [{ name: 'example-category', slug: 'example-category-slug' }]

const post: PostLinkProps = 
  { slug: 'example-test-post', categories: categories, title: 'example-test-tile' }

describe('<PostLink />', () => {
  test('renders correctly', () => {
    const { container } = render(<PostLink {...post} />);
    expect(container).toMatchSnapshot();
  })
})
