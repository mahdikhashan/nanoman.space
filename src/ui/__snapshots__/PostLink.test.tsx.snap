// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostLink /> renders correctly 1`] = `
<div>
  <a
    class="flex flex-row py-4 justify-between hover:text-orange-500 transition ease-in-out delay-50 hover:scale-[1.025]"
    href="/blog/example-test-post"
  >
    <div>
      example-test-tile
    </div>
    <div
      class="text-zinc-500"
    >
      example-category
    </div>
  </a>
</div>
`;
